### PR TITLE
ds-333-v2-create-input-checkbox

### PR DESCRIPTION
**Pull request by bibifrost**
# Note
This PR replace the [old PR](https://github.com/bifrost-vui/bifrost-front-end-library/pull/327). Because it didn't keep history of modified files.

# What I did:

## General
- Create **Input Checkbox** and **Input Radio** components.
- Documente Input Anatomy in Figma:
image

## input-radio-option and input-checkbox-option
- Rename **input-checkbox** to **input-checkbox-option** and **input-radio** to **input-radio-option**.
- Fix display of native required message:
image
- Fix height of **input-radio-option**, **input-radio-checkbox**. Previously:
image
image

## input-radio-control and input-checkbox-control
- Rename CSS Class `.bf-checkbox` to `.bf-checkbox-control` and `.bf-radio` to `.bf-radio-control`.

## Selection Tile
- Unify **Selection Tile** API with **Input Checkbox Option**:
  * `inputId` => `id` (No breaking change)
  * `isSelected` => `isChecked` (Breaking change!)
  * `title` =>`label` (Breaking change!)

## Form example
- Create a page **Styles/Inputs** to check inputs behavior and output.
- Warning: Waiting for the PR submit Button. For instance, force locally `type=submit` in Button.

### CODE REVIEW

File: projects/front-end-library/src/lib/components/selection-tile/angular/api.model.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    iconName: string;
+    isRequired: boolean;
+    isDisabled: boolean;
+    isInvalid: boolean;
+    errorMessage: string;
+    ariaLabel: string;
+    orientation: 'horizontal' | 'vertical';
+    fit: 'content' | 'container';
+    content: string | 'TwigBlock';
+    extraAttribute: string;
```
Comment (JonathanLefebvreV) [inline projects/front-end-library/src/lib/components/selection-tile/angular/api.model.ts:20]: footer property is missing from this interface. It's the most expected feature of the Input List for the current project that wait for us.

File: projects/front-end-library/src/lib/components/input-radio/twig/index.twig [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      <ul class='bf-input-radio__options list-unstyled {{ (areOptionTiles and areOptionTilesInline) ? 'bf-input-radio__options--inline' : 'd-flex flex-column flex-wrap' }} gap-2 mb-0'>
+        {% if areOptionTiles|default(false) %}
+          {% for option in options %}
+            {% set option = option|merge({
+              'type': 'radio',
+              'name': name,
+              'isRequired': option.isRequired|default(isRequired)|default(false),
+              'orientation': areOptionTilesInline ? 'vertical' : 'horizontal',
+              'class': option.class|default('') ~ (areOptionTilesLarge ? ' bf-selection-tile--large' : ''),
+              'fit': 'content',
```
Comment (JonathanLefebvreV) [inline projects/front-end-library/src/lib/components/input-radio/twig/index.twig:18]: Note to myself: need to test it more. Seem a hack that skips a feature of the selectionTile. 
I don't understand why it is not managed as a dynamic property

**Review by JonathanLefebvreV — CHANGES_REQUESTED** [review]

**Comment by JonathanLefebvreV** [comment]
Note: if times before the packaging. 
- Refactor https://bifrost-front-end-library-vui.vercel.app/?path=/story/components-selection-tile--drupal-example-selection-color to be in the input list example instead

File: projects/front-end-library/src/lib/components/selection-tile/angular/api.model.ts (hunk 1 above)
Comment (bibifrost) [reply projects/front-end-library/src/lib/components/selection-tile/angular/api.model.ts:20]: Done

**Comment by bibifrost** [comment]
> Note: if times before the packaging.
> 
> * Refactor https://bifrost-front-end-library-vui.vercel.app/?path=/story/components-selection-tile--drupal-example-selection-color to be in the input list example instead

Done

File: projects/front-end-library/src/lib/components/input-radio/twig/index.twig (hunk 2 above)
Comment (bibifrost) [reply projects/front-end-library/src/lib/components/input-radio/twig/index.twig:18]: C'est volontaire, car container = size large (avec un min-width et min-height).
Or, vu avec Kathleen, ce cas n'est pas désiré car on veux que toutes les tuiles aient la meme largeur pour faire des colonnes parfaites. C'est pourquoi `display: grid` prend le dessus et définie la largeur des tuiles.
Il existe cependant la props `areOptionTilesLarge` qui permet de grandir la taille des tuiles.

**Comment by JonathanLefebvreV** [comment]
Merci @bibifrost d'avoir fait les fix. Tu viens de rendre ma journée moins intense. 
Je commence le 2e review de tout ça

**Review by JonathanLefebvreV — APPROVED** [review]

**Pull request merged**